docs/source/design/hyspecpptclasses.rst

```diff
@@ -136,3 +446,205 @@ HyspecPPT Presenter
+            Note left of View: crosshair_parameters_update is triggered
+
+
+
+#. Display the available plot types from the settings files: set_plot_options() at the View
+
+    .. mermaid::
+
+        sequenceDiagram
+            participant View
+            participant Presenter
+
+            Note over View,Presenter: Application Start - HyspecPPTView Initialization
+            View->>Presenter: Get all available plot type options - ExperimentWidget::get_plot_options()
+            Note right of Presenter: get the PlotType Enum from experiment_settings file
+            Presenter->>View: Return the list of plot types (str)
+            Note left of View: Set and display the plot types in the plot_type_value combo box
+
+#. Display the available experiment type options from the settings files: set_experiment_type_options() at the View
+
+    .. mermaid::
+
+        sequenceDiagram
+            participant View
+            participant Presenter
+
+            Note over View,Presenter: Application Start
+            View->>Presenter: Get all available experiment type options - ExperimentWidget::get_experiment_type_options()
+            Note right of Presenter: get the ExperimentType Enum from experiment_settings file
+            Presenter->>View: Return the list of experiment types (str)
+            Note left of View: Set and display the experiment types in the experiment_type_value radio buttons
+
+#. This describes the sequence of events happening among M-V-P when Experiment parameters are updated in order to see a new plot : experiment_parameters_update()
+
+    * Valid Status:
+
+        .. mermaid::
+
+            sequenceDiagram
+                participant View
+                participant Presenter
+                participant Model
+
+                Note over View,Model: Plot draw due to any ExperimentWidget parameter update
+                View->>Presenter: User updates a parameter at ExperimentWidget: ei_value, s2_value, p_value or plot_type_value
+                Note right of Presenter: Check the validation status of all ExperimentWidget parameters (ExperimentWidget.validation_status)
+                Presenter->>View: Gather the ExperimentWidget parameters (ExperimentWidget.get_parameters)
+                Presenter->>Model: Send the parameters to calculate plot (Experiment.calculate_graph_data)
+                Note right of Model: Store the ei, s2 p and plot_type in Experiment (Experiment.store_data internally) and calculate plot data
+                Model->>Presenter: Return graph data dictionary
+                Presenter->>View: Return graph data (HyspecPPTView.update_plot)
+                Note left of View: Draw the plot
+
+    * Invalid Status:
+
+    .. mermaid::
+
+        sequenceDiagram
+            participant View
+            participant Presenter
+            participant Model
+
+            Note over View,Model: Crosshair update due to any ExperimentWidget parameter update
```

Comment:
Note is wrong (it should be the same as the valid status

Reply:
Which line do you mean should be the same as valid status? Line 535? or the entire note?

Reply:
"Crosshair update due to any ExperimentWidget parameter update" should be "Plot draw due to any ExperimentWidget parameter update"